In [1]:
import pandas as pd
import numpy as np

In [3]:
import seaborn as sns 
import matplotlib.pyplot as plt
%matplotlib inline


In [ ]:
!pip install plotly
# interactive visualization

In [39]:
import plotly.express as px
import plotly.graph_objs as go
# import plotly.figure_factory as ff
from plotly.subplots import make_subplots
# for offline ploting
from plotly.offline import plot, iplot, init_notebook_mode
init_notebook_mode(connected=True)

In [40]:
# converter
# from pandas.plotting import register_matplotlib_converters
# register_matplotlib_converters()   

# hide warnings
import warnings
warnings.filterwarnings('ignore')

# to USA states details
# import us

In [41]:
# color pallette
cnf, dth, rec, act = '#393e46', '#ff2e63', '#21bf73', '#fe9801' 

# seaborn plot style
# sns.set_style('darkgrid')

In [9]:
country=pd.read_csv('country_wise_latest.csv')


In [47]:
country['Country/Region' == 'Republic Korea']

KeyError: False

In [18]:
country.rename(columns = {'Country/Region' : 'Country'}, inplace = True)


In [19]:
country.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country                 187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    float64
 9   Recovered / 100 Cases   187 non-null    float64
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    float64
 14  WHO Region              187 non-null    ob

In [14]:
country.describe()

Confirmed         Deaths     Recovered        Active     New cases  \
count  1.870000e+02     187.000000  1.870000e+02  1.870000e+02    187.000000   
mean   8.813094e+04    3497.518717  5.063148e+04  3.400194e+04   1222.957219   
std    3.833187e+05   14100.002482  1.901882e+05  2.133262e+05   5710.374790   
min    1.000000e+01       0.000000  0.000000e+00  0.000000e+00      0.000000   
25%    1.114000e+03      18.500000  6.265000e+02  1.415000e+02      4.000000   
50%    5.059000e+03     108.000000  2.815000e+03  1.600000e+03     49.000000   
75%    4.046050e+04     734.000000  2.260600e+04  9.149000e+03    419.500000   
max    4.290259e+06  148011.000000  1.846641e+06  2.816444e+06  56336.000000   

        New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
count   187.000000     187.000000          187.000000             187.000000   
mean     28.957219     933.812834            3.019519              64.820535   
std     120.037173    4197.719635            3.454302              26.287694   
min       0.000000       0.000000            0.000000               0.000000   
25%       0.000000       0.000000            0.945000              48.770000   
50%       1.000000      22.000000            2.150000              71.320000   
75%       6.000000     221.000000            3.875000              86.885000   
max    1076.000000   33728.000000           28.560000             100.000000   

       Deaths / 100 Recovered  Confirmed last week  1 week change  \
count                  187.00         1.870000e+02     187.000000   
mean                      inf         7.868248e+04    9448.459893   
std                       NaN         3.382737e+05   47491.127684   
min                      0.00         1.000000e+01     -47.000000   
25%                      1.45         1.051500e+03      49.000000   
50%                      3.62         5.020000e+03     432.000000   
75%                      6.44         3.708050e+04    3172.000000   
max                       inf         3.834677e+06  455582.000000   

       1 week % increase  
count         187.000000  
mean           13.606203  
std            24.509838  
min            -3.840000  
25%             2.775000  
50%             6.890000  
75%            16.855000  
max           226.320000

In [8]:
day = pd.read_csv('day_wise.csv')
day

Date  Confirmed  Deaths  Recovered   Active  New cases  New deaths  \
0    2020-01-22        555      17         28      510          0           0   
1    2020-01-23        654      18         30      606         99           1   
2    2020-01-24        941      26         36      879        287           8   
3    2020-01-25       1434      42         39     1353        493          16   
4    2020-01-26       2118      56         52     2010        684          14   
..          ...        ...     ...        ...      ...        ...         ...   
183  2020-07-23   15510481  633506    8710969  6166006     282756        9966   
184  2020-07-24   15791645  639650    8939705  6212290     281164        6144   
185  2020-07-25   16047190  644517    9158743  6243930     255545        4867   
186  2020-07-26   16251796  648621    9293464  6309711     204606        4104   
187  2020-07-27   16480485  654036    9468087  6358362     228693        5415   

     New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0                0                3.06                   5.05   
1                2                2.75                   4.59   
2                6                2.76                   3.83   
3                3                2.93                   2.72   
4               13                2.64                   2.46   
..             ...                 ...                    ...   
183         169714                4.08                  56.16   
184         228736                4.05                  56.61   
185         219038                4.02                  57.07   
186         134721                3.99                  57.18   
187         174623                3.97                  57.45   

     Deaths / 100 Recovered  No. of countries  
0                     60.71                 6  
1                     60.00                 8  
2                     72.22                 9  
3                    107.69                11  
4                    107.69                13  
..                      ...               ...  
183                    7.27               187  
184                    7.16               187  
185                    7.04               187  
186                    6.98               187  
187                    6.91               187  

[188 rows x 12 columns]

In [11]:
full_grouped = pd.read_csv('full_grouped.csv')

In [12]:
full_grouped

Date      Country/Region  Confirmed  Deaths  Recovered  Active  \
0      2020-01-22         Afghanistan          0       0          0       0   
1      2020-01-22             Albania          0       0          0       0   
2      2020-01-22             Algeria          0       0          0       0   
3      2020-01-22             Andorra          0       0          0       0   
4      2020-01-22              Angola          0       0          0       0   
...           ...                 ...        ...     ...        ...     ...   
35151  2020-07-27  West Bank and Gaza      10621      78       3752    6791   
35152  2020-07-27      Western Sahara         10       1          8       1   
35153  2020-07-27               Yemen       1691     483        833     375   
35154  2020-07-27              Zambia       4552     140       2815    1597   
35155  2020-07-27            Zimbabwe       2704      36        542    2126   

       New cases  New deaths  New recovered             WHO Region  
0              0           0              0  Eastern Mediterranean  
1              0           0              0                 Europe  
2              0           0              0                 Africa  
3              0           0              0                 Europe  
4              0           0              0                 Africa  
...          ...         ...            ...                    ...  
35151        152           2              0  Eastern Mediterranean  
35152          0           0              0                 Africa  
35153         10           4             36  Eastern Mediterranean  
35154         71           1            465                 Africa  
35155        192           2             24                 Africa  

[35156 rows x 10 columns]

In [44]:
full_grouped.groupby('Country/Region').sum()

Confirmed  Deaths  Recovered   Active  New cases  \
Country/Region                                                         
Afghanistan           1936390   49098     798240  1089052      36263   
Albania                196702    5708     118877    72117       4880   
Algeria               1179755   77972     755897   345886      27973   
Andorra                 94404    5423      69074    19907        907   
Angola                  22662    1078       6573    15011        950   
...                       ...     ...        ...      ...        ...   
West Bank and Gaza     233461    1370      61124   170967      10621   
Western Sahara            901      63        648      190         10   
Yemen                   67180   17707      23779    25694       1691   
Zambia                 129421    2643      83611    43167       4552   
Zimbabwe                50794     881      12207    37706       2710   

                    New deaths  New recovered  
Country/Region                                 
Afghanistan               1269          25198  
Albania                    144           2745  
Algeria                   1163          18837  
Andorra                     52            803  
Angola                      41            242  
...                        ...            ...  
West Bank and Gaza          78           3752  
Western Sahara               1              8  
Yemen                      483            833  
Zambia                     140           2815  
Zimbabwe                    36            542  

[187 rows x 7 columns]

In [17]:
country_data = pd.read_csv('countries of the world.csv')
country_data

Country                               Region  Population  \
0       Afghanistan         ASIA (EX. NEAR EAST)             31056997   
1           Albania   EASTERN EUROPE                          3581655   
2           Algeria   NORTHERN AFRICA                        32930091   
3    American Samoa   OCEANIA                                   57794   
4           Andorra   WESTERN EUROPE                            71201   
..               ...                                  ...         ...   
222       West Bank   NEAR EAST                               2460492   
223  Western Sahara   NORTHERN AFRICA                          273008   
224           Yemen   NEAR EAST                              21456188   
225          Zambia   SUB-SAHARAN AFRICA                     11502010   
226        Zimbabwe   SUB-SAHARAN AFRICA                     12236805   

     Area (sq. mi.) Pop. Density (per sq. mi.) Coastline (coast/area ratio)  \
0            647500                       48,0                         0,00   
1             28748                      124,6                         1,26   
2           2381740                       13,8                         0,04   
3               199                      290,4                        58,29   
4               468                      152,1                         0,00   
..              ...                        ...                          ...   
222            5860                      419,9                         0,00   
223          266000                        1,0                         0,42   
224          527970                       40,6                         0,36   
225          752614                       15,3                         0,00   
226          390580                       31,3                         0,00   

    Net migration Infant mortality (per 1000 births)  GDP ($ per capita)  \
0           23,06                             163,07               700.0   
1           -4,93                              21,52              4500.0   
2           -0,39                                 31              6000.0   
3          -20,71                               9,27              8000.0   
4             6,6                               4,05             19000.0   
..            ...                                ...                 ...   
222          2,98                              19,62               800.0   
223           NaN                                NaN                 NaN   
224             0                               61,5               800.0   
225             0                              88,29               800.0   
226             0                              67,69              1900.0   

    Literacy (%) Phones (per 1000) Arable (%) Crops (%) Other (%) Climate  \
0           36,0               3,2      12,13      0,22     87,65       1   
1           86,5              71,2      21,09      4,42     74,49       3   
2           70,0              78,1       3,22      0,25     96,53       1   
3           97,0             259,5         10        15        75       2   
4          100,0             497,2       2,22         0     97,78       3   
..           ...               ...        ...       ...       ...     ...   
222          NaN             145,2       16,9     18,97     64,13       3   
223          NaN               NaN       0,02         0     99,98       1   
224         50,2              37,2       2,78      0,24     96,98       1   
225         80,6               8,2       7,08      0,03      92,9       2   
226         90,7              26,8       8,32      0,34     91,34       2   

    Birthrate Deathrate Agriculture Industry Service  
0        46,6     20,34        0,38     0,24    0,38  
1       15,11      5,22       0,232    0,188   0,579  
2       17,14      4,61       0,101      0,6   0,298  
3       22,46      3,27         NaN      NaN     NaN  
4        8,71      6,25         NaN      NaN     NaN  
..        ...      

In [22]:
country_all = pd.merge(country,country_data,how = 'inner', on='Country')

In [25]:
country_all = pd.concat([country, country_data], axis=1, join='inner')
country_all

Country  Confirmed  Deaths  Recovered  Active  New cases  \
0           Afghanistan      36263    1269      25198    9796        106   
1               Albania       4880     144       2745    1991        117   
2               Algeria      27973    1163      18837    7973        616   
3               Andorra        907      52        803      52         10   
4                Angola        950      41        242     667         18   
..                  ...        ...     ...        ...     ...        ...   
182  West Bank and Gaza      10621      78       3752    6791        152   
183      Western Sahara         10       1          8       1          0   
184               Yemen       1691     483        833     375         10   
185              Zambia       4552     140       2815    1597         71   
186            Zimbabwe       2704      36        542    2126        192   

     New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0            10             18                3.50                  69.49   
1             6             63                2.95                  56.25   
2             8            749                4.16                  67.34   
3             0              0                5.73                  88.53   
4             1              0                4.32                  25.47   
..          ...            ...                 ...                    ...   
182           2              0                0.73                  35.33   
183           0              0               10.00                  80.00   
184           4             36               28.56                  49.26   
185           1            465                3.08                  61.84   
186           2             24                1.33                  20.04   

     ...  Phones (per 1000)  Arable (%)  Crops (%)  Other (%) Climate  \
0    ...                3,2       12,13       0,22      87,65       1   
1    ...               71,2       21,09       4,42      74,49       3   
2    ...               78,1        3,22       0,25      96,53       1   
3    ...              259,5          10         15         75       2   
4    ...              497,2        2,22          0      97,78       3   
..   ...                ...         ...        ...        ...     ...   
182  ...              262,4        2,22      13,33      84,45       2   
183  ...                4,0        6,98       0,89      92,13       2   
184  ...              411,4        1,64          0      98,36       2   
185  ...              220,1       30,16       2,62      67,22       3   
186  ...              406,1         8,6       1,49      89,91     NaN   

    Birthrate Deathrate  Agriculture  Industry Service  
0        46,6     20,34         0,38      0,24    0,38  
1       15,11      5,22        0,232     0,188   0,579  
2       17,14      4,61        0,101       0,6   0,298  
3       22,46      3,27          NaN       NaN     NaN  
4        8,71      6,25          NaN       NaN     NaN  
..        ...       ...          ...       ...     ...  
182     16,03      6,29        0,032     0,304   0,665  
183     45,76     23,03         0,49      0,31    0,21  
184      9,34      4,28            0     0,339   0,661  
185     10,65      9,45        0,035     0,294   0,672  
186      8,98     10,31        0,028     0,369   0,603  

[187 rows x 35 columns]

# 전 세계 187개 국가의 코로나 19 감염 현황에 대한 2020년 7월 27일까지의 데이터

In [29]:
country_all.to_csv('country_all.csv')

In [36]:
country.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country                 187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    float64
 9   Recovered / 100 Cases   187 non-null    float64
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    float64
 14  WHO Region              187 non-null    ob

In [54]:
temp = day[['Date','Deaths', 'Recovered', 'Active']].tail(1)
temp = temp.melt(id_vars="Date", value_vars=['Active', 'Deaths', 'Recovered'])
fig = px.treemap(temp, path=["variable"], values="value", height=225, 
                 color_discrete_sequence=[act, rec, dth])
fig.data[0].textinfo = 'label+text+value'
fig.show()

In [57]:
temp = country[['Country','Deaths', 'Recovered', 'Active']].tail(1)
temp = temp.melt(id_vars="Country", value_vars=['Active', 'Deaths', 'Recovered'])
fig = px.treemap(temp, path=["variable"], values="value", height=225, 
                 color_discrete_sequence=[act, rec, dth])
fig.data[0].textinfo = 'label+text+value'
fig.show()

In [61]:
def plot_map(df, col, pal):
    df = df[df[col]>0]
    fig = px.choropleth(df, locations="Country", locationmode='country names', 
                  color=col, hover_name="Country", 
                  title=col, hover_data=[col], color_continuous_scale=pal)
#     fig.update_layout(coloraxis_showscale=False)
    fig.show()


In [62]:
plot_map(country, 'Confirmed', 'matter')

In [63]:
plot_map(country, 'Deaths', 'matter')

In [64]:
plot_map(country, 'Recovered', 'matter')

In [66]:
plot_map(country, 'Active', 'matter')